src/temporal_logic/rules.rs

```diff
@@ -1175,6 +1223,42 @@ pub proof fn wf1_assume<T>(spec: TempPred<T>, next: ActionPred<T>, forward: Acti
     wf1_variant_assume_temp::<T>(spec, lift_action(next), lift_action(forward), lift_state(asm), lift_state(p), lift_state(q));
 }
 
+pub proof fn leads_to_confluence_self_temp<T>(spec: TempPred<T>, next: TempPred<T>, p: TempPred<T>, q: TempPred<T>)
```

Comment:
I guess we will not need this lemma if you can simplify `lemma_reconcile_idle_leads_to_reconcile_idle_and_scheduled_by_assumption` as mentioned above. But let's keep the lemmas for now.